endpoints/openrtb2/amp_auction.go

```diff
@@ -363,9 +366,25 @@ func (deps *endpointDeps) overrideWithParams(httpRequest *http.Request, req *ope
 		req.Imp[0].TagID = slot
 	}
 
+	//In the AMP endpoint the consent string found in the http.Request query overrides that of the prebid query
+	queryConsentString := httpRequest.FormValue("gdpr_consent")
+	if queryConsentString != "" {
+		// If nil, initialize
+		if req.User == nil {
+			req.User = &openrtb.User{}
+		}
+		jsonMsg, err := json.Marshal(&openrtb_ext.ExtUser{Consent: queryConsentString})
+		if err != nil {
+			return err
+		}
+		req.User.Ext = jsonMsg
```

Comment:
This will overwrite anything else that exists in `user.ext` One possibility is to edit the raw json with jsonparser, which is already part of the project. Another option might be to use json-patch, which we already use for resolving stored requests.

Reply:
Done but used the `regexp` library instead.